### ISSUE TITLE
Allow attributes to be placed above constructor properties

### ISSUE THREAD

**Issue by piotrjoniec** [prettier/plugin-php#2329]
Currently, attributes used on constructor properties are forcefully placed inline with the property (or above if it doesn't fit the print width).

This can result in chaotic code where the types and names of the properties are all over the place, making it much harder to scan.

I feel like it would be better to allow the attributes to retain their original placement, as long as everything fits within the print width.

Here's an example of this happening:

**@prettier/plugin-php v0.22.2**
[Playground link](https://loilo.github.io/prettier-php-playground/#N4IgDgTgpgLjCWUIHkwIPYDsDOIBcok8mMA6vACYwAW+AHAAwA0IMAhgEblW14AsLAK7YoAFU648AMzYAbES2zEA5rKgBFQehhR8MCIKgsw1MADUkSrPhB0AdACYQLfW3iyVAYXQBbH2wAFAAkAvQMjEA4INgBjKABlGABPNRswJABaGNwWaABHQXhoAOjlf3wZeQjiEQgYErYytgq5EQBfFigKeBh0FDR4LElQNmwYFqqOkGIwQXG8EAAeAH4TMAAdTE2Y2VHsAAIzNgh4NhIAETZ2fagADx1MCgPL9k3gTf3P-dmODxj9qSCTAxDCYfYAfXBMSG+kEIIAFB8vsiAMQAbQAord4GNsPCjiczjA8HgdnsmPsAOSUSkASgAukjkZ8fn99sQYAAfVCguT7AAklCYmyZzNZ8H+HM5mEEslkAp8XVO4KFIrBzM+6IASlACkUuuQaOg5vDqRQ6RSAGLuNQUCmJE6YZSiJLpCkAWTYt3hDgArL6GaLkeL-mNHcoBQgYGphVt1RrtbrCtAKIbqMaYKaabSKQA5QSKk4xD3EeEMQPxsWCX4S-ZhlQCohxWNBr7o-OFiUlzBlisalnVtn1p3S2Xy-nQnxgY5QcFXcFNoxq-v7dHu0vlxmV4OD2schXEcFjdAxADWseRtP270rbU2d8wIDaQA)

**Input:**
```php
<?php

class VariantData extends Data
{
    public function __construct(
        #[Exists(Variant::class, 'id')]
        public int|Optional $id,

        public int|null $media_id,

        #[RequiredWithout('id'), Filled, StringType, Max(255)]
        public string $title,

        #[RequiredWithout('id'), Numeric, Min(0)]
        public string $price,

        #[Numeric, Min(0)]
        public string|null $compare_at_price,

        #[Min(0)]
        public int $min_stock,
    ) {
    }
}

```

**Output:**
```php
<?php

class VariantData extends Data
{
    public function __construct(
        #[Exists(Variant::class, 'id')] public int|Optional $id,

        public int|null $media_id,

        #[
            RequiredWithout('id'),
            Filled,
            StringType,
            Max(255),
        ]
        public string $title,

        #[RequiredWithout('id'), Numeric, Min(0)] public string $price,

        #[Numeric, Min(0)] public string|null $compare_at_price,

        #[Min(0)] public int $min_stock,
    ) {
    }
}

```

**Comment by cseufert** [comment]
Yes, appears that this would be in line with the latest PER CS for Attributes
https://www.php-fig.org/per/coding-style/#12-attributes

Following the spec it looks like all attributes should be on their own line. I am not sure if we want this in closures parameters though.
